pkg/informer/informer.go

```diff
@@ -87,14 +104,40 @@ type DynamicDiscoverySharedInformerFactory struct {
 	subscribers     map[string]chan<- struct{}
 }
 
-// NewDynamicDiscoverySharedInformerFactory returns a factory for shared
-// informers that discovers new types and informs on updates to resources of
+// NewDynamicDiscoverySharedInformerFactory returns a factory for cluster-aware
+// shared informers that discovers new types and informs on updates to resources of
 // those types.
+// It retrieves GVR-related information from CRD informers.
 func NewDynamicDiscoverySharedInformerFactory(
 	cfg *rest.Config,
 	filterFunc func(obj interface{}) bool,
 	crdInformer kcpapiextensionsv1informers.CustomResourceDefinitionClusterInformer,
 	indexers cache.Indexers,
+) (*DynamicDiscoverySharedInformerFactory, error) {
+	crdIndexer := crdInformer.Informer().GetIndexer()
+
+	// Add an index function that indexes a CRD by its group/version/resource.
+	if err := crdInformer.Informer().AddIndexers(cache.Indexers{
+		byGroupVersionResourceIndex: byGroupVersionResourceIndexFunc,
+	}); err != nil {
+		return nil, err
+	}
+
+	return NewDelegatingDynamicDiscoverySharedInformerFactory(cfg, filterFunc, indexers, &crdGVRSource{
+		crdInformer: crdInformer,
+		crdIndexer:  crdIndexer,
+	})
+}
+
+// NewDelegatingDynamicDiscoverySharedInformerFactory returns a factory for cluster-aware
```

Comment:
I don't think `Delegating` gives much information to the user here - the shared informer factory doesn't delegate it's operation

Reply:
true